src/vs/workbench/contrib/comments/browser/commentReply.ts

```diff
@@ -77,32 +75,44 @@ export class CommentReply<T extends IRange | ICellRange> extends Disposable {
 		this.commentEditorIsEmpty = CommentContextKeys.commentIsEmpty.bindTo(this._contextKeyService);
 		this.commentEditorIsEmpty.set(!this._pendingComment);
 
+		const formActions = dom.append(this.form, dom.$('.form-actions'));
+		this._formActions = dom.append(formActions, dom.$('.other-actions'));
+		this._editorActions = dom.append(formActions, dom.$('.editor-actions'));
+		this.initialize();
+	}
+
+	async initialize() {
 		const hasExistingComments = this._commentThread.comments && this._commentThread.comments.length > 0;
 		const modeId = generateUuid() + '-' + (hasExistingComments ? this._commentThread.threadId : ++INMEM_MODEL_ID);
 		const params = JSON.stringify({
 			extensionId: this._commentThread.extensionId,
 			commentThreadId: this._commentThread.threadId
 		});
 
-		let resource = URI.parse(`${COMMENT_SCHEME}://${this._commentThread.extensionId}/commentinput-${modeId}.md?${params}`); // TODO. Remove params once extensions adopt authority.
-		const commentController = this.commentService.getCommentController(owner);
+		let resource = URI.from({
+			scheme: Schemas.commentsInput,
+			path: `/${this._commentThread.extensionId}/commentinput-${modeId}.md?${params}` // TODO. Remove params once extensions adopt authority.
+		});
+		const commentController = this.commentService.getCommentController(this.owner);
 		if (commentController) {
 			resource = resource.with({ authority: commentController.id });
 		}
 
-		const model = this.modelService.createModel(this._pendingComment || '', this.languageService.createByFilepathOrFirstLine(resource), resource, false);
+		const model = await this.textModelService.createModelReference(resource);
```

Comment:
Why do we need to do this instead of directly creating the model and keeping it synch? The comment editor is already visible to extensions ([for example](https://github.com/microsoft/vscode-pull-request-github/blob/b6cbab4a1428bc69b48523886d26d00df71721ac/src/commands.ts#L1039) as the `activeTextEditor`)

Reply:
My understanding is that this is required for applying workspace edits. Here's the relevant code:

https://github.com/microsoft/vscode/blob/826b140ba48432a629e813d988e753fe233765e8/src/vs/workbench/contrib/bulkEdit/browser/bulkTextEdits.ts#L231

Code actions and paste use workspace edits. Without this, I saw warnings about being unable to resolve the reference file

Reply:
Thanks for the explanation!